### ISSUE TITLE
[Bug]: TagPicker (react-tag-picker-preview) - Keyborg instance k* is being disposed incorrectly

### ISSUE THREAD

**Issue by pradeept95** [microsoft/fluentui#30984]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
```

### Are you reporting Accessibility issue?
Yes

### Reproduction

https://stackblitz.com/edit/vitejs-vite-pdjx5p?file=src%2FExample.tsx

## Actual Behavior
When I create a new project and test TagPicker, it throws `Keyborg instance k10 is being disposed incorrectly` when close the picker list. Also noticed that, when I start typing in picker input, it does not highlight any item even though it matches in the list like in the docs site. 
**Here is how it looks in my example:**
![image](https://github.com/microsoft/fluentui/assets/28040299/c285c78d-319f-4443-92bc-58378e961843)

**Here is how it looks in docs site:**

![image](https://github.com/microsoft/fluentui/assets/28040299/8e4fec52-57bc-4537-ba56-49318ef25a48)

Here are the packages I am using: 
    "@fluentui/react-components": "^9.47.5",
    "@fluentui/react-tag-picker-preview": "^0.2.0",

Could anyone help me with what I might have done wrong here? 

## Expected Behavior
Should not throw error and keyboard navigation should work

### Logs

```shell
Keyborg instance k10 is being disposed incorrectly.
dispose @ chunk-KCJA4Q5K.js?v=83a9c81a:5072
dispose @ chunk-KCJA4Q5K.js?v=83a9c81a:5054
disposeKeyborg @ chunk-KCJA4Q5K.js?v=83a9c81a:5115
(anonymous) @ chunk-KCJA4Q5K.js?v=83a9c81a:13679
safelyCallDestroy @ chunk-FQB4DAA4.js?v=83a9c81a:16735
commitHookEffectListUnmount @ chunk-FQB4DAA4.js?v=83a9c81a:16862
commitPassiveUnmountInsideDeletedTreeOnFiber @ chunk-FQB4DAA4.js?v=83a9c81a:18277
commitPassiveUnmountEffectsInsideOfDeletedTree_begin @ chunk-FQB4DAA4.js?v=83a9c81a:18239
commitPassiveUnmountEffects_begin @ chunk-FQB4DAA4.js?v=83a9c81a:18175
commitPassiveUnmountEffects @ chunk-FQB4DAA4.js?v=83a9c81a:18163
flushPassiveEffectsImpl @ chunk-FQB4DAA4.js?v=83a9c81a:19483
flushPassiveEffects @ chunk-FQB4DAA4.js?v=83a9c81a:19441
commitRootImpl @ chunk-FQB4DAA4.js?v=83a9c81a:19410
commitRoot @ chunk-FQB4DAA4.js?v=83a9c81a:19271
performSyncWorkOnRoot @ chunk-FQB4DAA4.js?v=83a9c81a:18889
flushSyncCallbacks @ chunk-FQB4DAA4.js?v=83a9c81a:9133
(anonymous) @ chunk-FQB4DAA4.js?v=83a9c81a:18621
chunk-KCJA4Q5K.js?v=83a9c81a:5072 Keyborg instance k14 is being disposed incorrectly.
dispose @ chunk-KCJA4Q5K.js?v=83a9c81a:5072
dispose @ chunk-KCJA4Q5K.js?v=83a9c81a:5054
disposeKeyborg @ chunk-KCJA4Q5K.js?v=83a9c81a:5115
(anonymous) @ chunk-KCJA4Q5K.js?v=83a9c81a:13679
safelyCallDestroy @ chunk-FQB4DAA4.js?v=83a9c81a:16735
commitHookEffectListUnmount @ chunk-FQB4DAA4.js?v=83a9c81a:16862
commitPassiveUnmountInsideDeletedTreeOnFiber @ chunk-FQB4DAA4.js?v=83a9c81a:18277
commitPassiveUnmountEffectsInsideOfDeletedTree_begin @ chunk-FQB4DAA4.js?v=83a9c81a:18239
commitPassiveUnmountEffects_begin @ chunk-FQB4DAA4.js?v=83a9c81a:18175
commitPassiveUnmountEffects @ chunk-FQB4DAA4.js?v=83a9c81a:18163
flushPassiveEffectsImpl @ chunk-FQB4DAA4.js?v=83a9c81a:19483
flushPassiveEffects @ chunk-FQB4DAA4.js?v=83a9c81a:19441
commitRootImpl @ chunk-FQB4DAA4.js?v=83a9c81a:19410
commitRoot @ chunk-FQB4DAA4.js?v=83a9c81a:19271
performSyncWorkOnRoot @ chunk-FQB4DAA4.js?v=83a9c81a:18889
flushSyncCallbacks @ chunk-FQB4DAA4.js?v=83a9c81a:9133
(anonymous) @ chunk-FQB4DAA4.js?v=83a9c81a:18621
```

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by pradeept95** [comment]
FYI: It seems this is happening on react 18+, I was testing on react 17, the behavior is working as expected

**Comment by yavulan** [comment]
Seems like the same error is produced by Combobox and Dropdown also

**Comment by miroslavstastny** [comment]
Fixed in #31295
